vertx-opentelemetry/src/main/java/io/vertx/tracing/opentelemetry/OpenTelemetryUtil.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2011-2021 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.tracing.opentelemetry;
+
+import io.opentelemetry.api.trace.Span;
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+
+public class OpenTelemetryUtil {
+
+  static String ACTIVE_CONTEXT = "tracing.context";
+  static String ACTIVE_SPAN = "tracing.span";
+
+  /**
+   * Get the active span from the current {@link Context}
+   *
+   * @return a {@link Span} or null
+   */
+  public static Span getSpan() {
```

Comment:
This approach will generally not work well in apps since other instrumentation (think Lettuce, mysql) will be reading the current context from `io.opentelemetry.context.Context`, not using this method. You will need to implement a context bridge, similar to the gRPC one we have here

https://github.com/open-telemetry/opentelemetry-java/tree/main/context/src/otelInGrpcTest/java/io/opentelemetry/context

This would allow using both `Context.current()` and `Vertx.currentContext()` to access the same information about OpenTelemetry, and because it would be "otelInVertx", the opentelemetry context would be propagated correctly across Vertx's async boundaries.

Reply:
I've tried to figure out how it works, but TBH I'm completely lost, since I don't know how to implement properly `OpenTelemetryTracer` then, if i should rely on vert.x context or opentelemetry context. I prefer to keep as is now and revisit later, ok? I made all the OpenTelemetryUtil methods package private, to make sure nobody can mess it up with them.

I definitely appreciate some input from your side, since it's not really clear to me `ContextStorage` interacts with the rest of the architecture.

Reply:
That's a draft of the VertxContextStorageProvider (which breaks the rest of the code) https://github.com/slinkydeveloper/vertx-tracing/commit/fb39bd562b380d9750e0cb4b5fc0c0a0da292dc5

Reply:
Yeah following up in another PR makes seems fine to me. The test we'll want to work on, if I understand vertx at all since I've never used it before :P, is something like here

https://github.com/eclipse-vertx/vertx-tracing/pull/22/files#diff-14a52de14bffe8ce1d7e359b96cafd9538985c7b200ea61a73bed07c04c4e0b6R276

If inside a `vertx.runOnContext`, we can use the AWS SDK or gRPC instrumentation from the `opentelemetry-java-instrumentation` project and the trace has those client spans wired as children of the server spans without extra code. It should be possible with the bridge I can help figure it out as a followup.

Reply:
Yeah I see your point and makes sense to me